### ISSUE TITLE
(React) client library

### ISSUE THREAD

**Issue by maximiliancsuk** [goauthentik/authentik#4687]
Hello,

is there a client library that I can add to my (React) web application to integrate Authentik as its auth provider? Looking for something similar to what Keycloak is providing: [https://github.com/react-keycloak/react-keycloak/blob/master/packages/web/README.md#setup-keycloakprovider](https://github.com/react-keycloak/react-keycloak/blob/master/packages/web/README.md#setup-keycloakprovider)

Ideally, it would be a React component with which I wrap my application, like it's possible with Keycloak:
```
const keycloak = new Keycloak({
  "url": `${env("KEYCLOAK_URL")}`,
  "realm": env("KEYCLOAK_REALM"),
  "clientId": env("KEYCLOAK_CLIENT_ID"),
  // other stuff
})
<ReactKeycloakProvider authClient={keycloak} onTokens={tokenSetter}>
  <MyApplication />
</ReactKeycloakProvider>
```

I was also looking through the more general OAuth2.0 libraries, but they were either framework specific ([NextAuth](https://next-auth.js.org/)) or badly maintained.

Any pointers?

**Comment by MarkusGnigler** [comment]
Maybe plain implementation with something like [oidc-client-ts](https://github.com/authts/oidc-client-ts) or a wrapper [react-oidc-context](https://github.com/authts/react-oidc-context) could help?

I'm also researching on this topic. Every hint is appreciated

**Comment by BeryJu** [comment]
closed in favour of #4513, we'll most likely add examples for https://github.com/authts/oidc-client-ts
